src/odemis/driver/smaract.py

```diff
@@ -2644,6 +2644,8 @@ def __init__(self, name, role, locator, ref_on_init=False, axes=None, speed=1e-3
             else:
                 logging.warning("SA_CTL is not referenced. The device will not function until referencing occurs.")
 
+        self._focuser_pos_updater = RepeatingTimer(1.0, self._updatePosition)
+        self._focuser_pos_updater.start()
```

Comment:
You should also stop it in `terminate()`. You can do `self._update_position_timer.cancel()`.